src/Microsoft.AspNetCore.Razor.Language/Intermediate/IntermediateNodeWalker.cs

```diff
@@ -7,11 +7,11 @@ namespace Microsoft.AspNetCore.Razor.Language.Intermediate
 {
     public abstract class IntermediateNodeWalker : IntermediateNodeVisitor
     {
-        private readonly Stack<IntermediateNode> _ancestors = new Stack<IntermediateNode>();
+        private readonly List<IntermediateNode> _ancestors = new List<IntermediateNode>();
 
-        protected IEnumerable<IntermediateNode> Ancestors => _ancestors;
```

Comment:
Yes it does 👍 https://github.com/aspnet/Razor/blob/dev/test/Microsoft.AspNetCore.Razor.Language.Test/Intermediate/IntermediateNodeWalkerTest.cs#L68.